code/workspaces/infrastructure-api/src/routers/clustersRouter.js

```diff
@@ -0,0 +1,27 @@
+import { service } from 'service-chassis';
+import express from 'express';
+import { permissionTypes } from 'common';
+import clustersController from '../controllers/clustersController';
+import permissionMiddleware from '../auth/permissionMiddleware';
+import { paramProjectKeyValidator } from './commonValidators';
+
+const { projectPermissions: { PROJECT_KEY_STACKS_CREATE, PROJECT_KEY_STACKS_LIST } } = permissionTypes;
+
+const { errorWrapper } = service.middleware;
+
+const clustersRouter = express.Router();
+
+clustersRouter.post(
+  '/',
+  permissionMiddleware(PROJECT_KEY_STACKS_CREATE),
+  clustersController.clusterValidator(),
+  errorWrapper(clustersController.createCluster),
+);
+clustersRouter.get(
+  '/project/:projectKey',
```

Comment:
For the asset repo metadata I used a `?projectKey=<projectKey>` query to allow for obtaining by a specific project. I thought this resulted in a slightly more intuitive endpoint i.e. `GET /clusters?projectKey=testproj` rather than `GET /clusters/project/testproj`. Which do you think it preferable?

Reply:
Based on the filtering example in https://stackoverflow.blog/2020/03/02/best-practices-for-rest-api-design/, I think you're right - I'll change it.